tests/components/smlight/conftest.py

```diff
@@ -89,6 +91,21 @@ def mock_smlight_client(request: pytest.FixtureRequest) -> Generator[MagicMock]:
 
         api.cmds = AsyncMock(spec_set=CmdWrapper)
         api.set_toggle = AsyncMock()
+        api.sse = MagicMock(spec_set=sseClient)
+        api.sse.client = AsyncMock()
+        api._settings_callbacks = {}
+
+        def _mock_settings_cb(
+            setting: Settings, callback: Callable
+        ) -> Callable[[], None]:
+            api._settings_callbacks.setdefault(setting, callback)
+
+            def remove_callback():
+                api._settings_callbacks.pop(setting, None)
+
+            return remove_callback
```

Comment:
Instead, just fetch the function the library received via the mock call arguments. This is implemented in `knocki`

Reply:
ok I see. have implemented something similar.